### ISSUE TITLE
overriding a final property with a getter generates a warning

### ISSUE THREAD

**Issue by Andersmholmgren** [dart-lang/dev_compiler#421]
In the following
```
class A {
  final String foo;
  A(this.foo);
}

class B extends A {
  B(String foo) : super(foo);

  String get foo => 'foo';
}
```

`String get foo => 'foo';` generates a warning.

It doesn't complain if `B implements A`.

I've done this for a while and never had any problems. It would seem odd that this would be dissallowed in the language

**Comment by jmesserly** [comment]
Hey, thanks for the feedback. This is a duplicate of https://github.com/dart-lang/dev_compiler/issues/52 and https://github.com/dart-lang/sdk/issues/24928. See those issues for more background. 24928 is still open--we have not settled this question yet.

It is a hard one, because supporting this ends causes widespread readability cost (and possibly performance cost), even in places where the feature is not used. For ahead-of-time compilation it's very helpful to know what is a field vs what is a virtual property.

That said we could make it easier to declare a field `@virtual` or something along those lines.

**Comment by jmesserly** [comment]
BTW, the other reason for the warning: your instances of "B" will have a dummy storage slot that is never used. Sometimes folks don't realize that their objects will be bigger and have unused storage space.

**Comment by Andersmholmgren** [comment]
Actually in the cases I found I could simply change it to implements (which I should have done in the first place). 

I think the `@virtual` suggestion might be a good tradeoff

**Comment by Andersmholmgren** [comment]
Actually I take that back. One of my cases did need to extend and strikes me as a legit case

It makes use of the parent classes with a fallback implementation for when the field is null

```
  Option<CompletedAction<E>> get completedAction =>
      super.completedAction.orElse(() => children
          .map((c) => c.completedAction)
          .firstWhere((ca) => ca is Some, orElse: () => const None()));
```
